ios/tunnel/tunnel_api.go

```diff
@@ -0,0 +1,163 @@
+// 2. localhost:{PORT}/tunnels			to get a list of all tunnels
+func ServeTunnelInfo(tm *TunnelManager, port int) error {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/tunnel/", func(writer http.ResponseWriter, request *http.Request) {
+		udid := strings.TrimPrefix(request.URL.Path, "/tunnel/")
+		if len(udid) == 0 {
+			return
+		}
+		tunnels, err := tm.ListTunnels()
+		idx := slices.IndexFunc(tunnels, func(info Tunnel) bool {
+			return info.Udid == udid
+		})
+		if idx < 0 {
+			http.Error(writer, "", http.StatusNotFound)
+			return
+		}
+		t := tunnels[idx]
+
+		writer.Header().Add("Content-Type", "application/json")
+		enc := json.NewEncoder(writer)
+		err = enc.Encode(t)
+		if err != nil {
+			return
+		}
+	})
+	mux.HandleFunc("/tunnels", func(writer http.ResponseWriter, request *http.Request) {
+		tunnels, err := tm.ListTunnels()
+
+		writer.Header().Add("Content-Type", "application/json")
+		enc := json.NewEncoder(writer)
+		err = enc.Encode(tunnels)
+		if err != nil {
+			return
+		}
+	})
+	if err := http.ListenAndServe(fmt.Sprintf("127.0.0.1:%d", port), mux); err != nil {
+		return fmt.Errorf("ServeTunnelInfo: failed to start http server: %w", err)
+	}
+	return nil
+}
+
+func TunnelInfoForDevice(udid string, tunnelInfoPort int) (Tunnel, error) {
+	c := http.Client{
+		Timeout: 5 * time.Second,
+	}
+	res, err := c.Get(fmt.Sprintf("http://127.0.0.1:%d/tunnel/%s", tunnelInfoPort, udid))
+	if err != nil {
+		return Tunnel{}, fmt.Errorf("TunnelInfoForDevice: failed to get tunnel info: %w", err)
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return Tunnel{}, fmt.Errorf("TunnelInfoForDevice: failed to read body: %w", err)
+	}
+	var info Tunnel
+	err = json.Unmarshal(body, &info)
+	if err != nil {
+		return Tunnel{}, fmt.Errorf("TunnelInfoForDevice: failed to parse response: %w", err)
+	}
+	return info, nil
+}
+
+// TunnelManager
+type TunnelManager struct {
+	ts tunnelStarter
+	dl deviceLister
+	pm PairRecordManager
+
+	tunnels map[string]Tunnel
+}
+
+// NewTunnelManager creates a new TunnelManager instance for setting up device tunnels for all connected devices
+func NewTunnelManager(pm PairRecordManager) *TunnelManager {
+	return &TunnelManager{
+		ts:      manualPairingTunnelStart{},
+		dl:      deviceList{},
+		pm:      pm,
+		tunnels: map[string]Tunnel{},
+	}
+}
+
+// UpdateTunnels checks for connected devices and starts a new tunnel if needed
+// On device disconnects the tunnel resources get cleaned up
+func (m *TunnelManager) UpdateTunnels() error {
+	devices, err := m.dl.ListDevices()
+	if err != nil {
+		return fmt.Errorf("UpdateTunnels: failed to get list of devices: %w", err)
+	}
+	for _, d := range devices.DeviceList {
+		udid := d.Properties.SerialNumber
+		if _, exists := m.tunnels[udid]; exists {
+			continue
+		}
+		log.WithField("udid", udid).Info("start tunnel")
+		t, err := m.ts.StartTunnel(context.TODO(), d, m.pm)
```

Comment:
4yi:
lol, I was thinking "what is context.TODO(), and what is the difference between this and context.Background()??"
turns out, there is none:
```var (

	background = new(emptyCtx)

	todo       = new(emptyCtx)

)


// Background returns a non-nil, empty Context. It is never canceled, has no

// values, and has no deadline. It is typically used by the main function,

// initialization, and tests, and as the top-level Context for incoming

// requests.

func Background() Context {

	return background

}


// TODO returns a non-nil, empty Context. Code should use context.TODO when

// it's unclear which Context to use or it is not yet available (because the

// surrounding function has not yet been extended to accept a Context

// parameter).

func TODO() Context {

	return todo

}
```

Reply:
removed that part 🙂 we take the context as an argument now. However, we also create a new child context based on the one we get with 10s for tunnel creation. Should we also expose this as an argument?